src/models/sqlOutputContentProvider.ts

```diff
@@ -339,33 +340,28 @@ export class SqlOutputContentProvider implements vscode.TextDocumentContentProvi
         // Get the active text editor
         let activeTextEditor = this._vscodeWrapper.activeTextEditor;
 
-        // Check if the results window already exists
-        if (!this.doesResultPaneExist(resultsUri)) {
-            // Wrapper tells us where the new results pane should be placed
-            let resultPaneColumn = this.newResultPaneViewColumn(queryUri);
-
-            // Try and Open new window then reset focus back to the editor
-            vscode.commands.executeCommand('vscode.previewHtml', resultsUri, resultPaneColumn, paneTitle).then(() => {
-                // get the result pane text editor to determine which column it was shown in
-                let resultPaneTextEditor = this._vscodeWrapper.visibleEditors.find(
-                    editor => editor.document.uri.toString() === resultsUri);
-
-                // get the result pane column from the text editor
-                if (resultPaneTextEditor !== undefined) {
-                    resultPaneColumn = resultPaneTextEditor.viewColumn;
-                }
+        // Wrapper tells us where the new results pane should be placed
+        let resultPaneColumn = this.newResultPaneViewColumn(queryUri);
 
-                // only reset focus to the text editor if it's in a different column then the results window
-                if (resultPaneColumn !== undefined
-                    && resultPaneColumn !== activeTextEditor.viewColumn) {
-                    this._vscodeWrapper.showTextDocument(activeTextEditor.document, activeTextEditor.viewColumn);
-                }
-            }, err => {
-                // Output to console if an error occurs
-                Utils.logToOutputChannel(err);
+        // Check if the results window already exists
+        let panel = this._resultsPanes.get(resultsUri);
+        if (!panel) {
+            panel = vscode.window.createWebviewPanel(resultsUri, paneTitle, resultPaneColumn, {
+                retainContextWhenHidden: true,
```

Comment:
We should probably think if we actually want this. This has a stark impact on performance as noted by vscode. As it currently stands we built the application to not need this (replaying events).

Reply:
That's a good point. I didn't see any performance impact in testing but I'm not familiar enough with it to know when you start seeing those issues.

I think maybe it's best to add a setting to let the user control it. Do you think we should set it to `false` by default and let users enable it if they want and that way we can test it for a release or two before enabling it for good?